cmd/node_problem_detector.go

```diff
@@ -46,23 +47,27 @@ func main() {
 
 	npdo.ValidOrDie()
 
-	monitors := make(map[string]types.Monitor)
-	for _, config := range npdo.SystemLogMonitorConfigPaths {
-		if _, ok := monitors[config]; ok {
-			// Skip the config if it's duplicated.
-			glog.Warningf("Duplicated monitor configuration %q", config)
-			continue
-		}
-		monitors[config] = systemlogmonitor.NewLogMonitorOrDie(config)
+	// Initialize problem daemons.
+	problemDaemons, errors := problemdaemon.NewProblemDaemons(npdo.MonitorsConfigs)
+	for _, err := range errors {
+		glog.Errorf("Problem daemon initialization failed: %v", err)
+	}
+	if len(errors) != 0 && npdo.PanicOnProblemDaemonFailure {
```

Comment:
Why not just always panic if a problem daemon fails? Any concern about that?

Reply:
Sounds fair :)
Changed the behavior to just panic.